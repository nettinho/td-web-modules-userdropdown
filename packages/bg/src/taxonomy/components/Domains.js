import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Route, Switch, withRouter } from "react-router-dom";
import { Header, Icon, Segment } from "semantic-ui-react";
import { bindActionCreators, compose } from "redux";
import { connect } from "react-redux";
import { FormattedMessage } from "react-intl";
import routes from "../routes";
import { createDomain } from "../routines";
import DomainCards from "./DomainCards";
import DomainForm from "./DomainForm";
import DomainsActions from "./DomainsActions";

export const Domains = ({ domainsLoading, createDomain }) => {
  return (
    <Fragment>
      <Segment>
        <Header as="h2">
          <Icon circular name="sitemap" />
          <Header.Content>
            <FormattedMessage id="navigation.glossary.taxonomy" />
            <Header.Subheader>
              <FormattedMessage id="domains.subheader" />
            </Header.Subheader>
          </Header.Content>
        </Header>
        <Segment attached="bottom" loading={domainsLoading}>
          <DomainsActions />
          <Switch>
            <Route path={routes.DOMAINS} component={DomainCards} exact />
            <Route path={routes.DOMAINS_NEW} exact>
              <DomainForm handleSubmit={createDomain} />
            </Route>
          </Switch>
        </Segment>
      </Segment>
    </Fragment>
  );
};

Domains.propTypes = {
  domainsLoading: PropTypes.bool,
  createDomain: PropTypes.func.isRequired
};

const mapStateToProps = ({ domainsLoading }) => ({
  domainsLoading
});

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ createDomain }, dispatch)
});

export default compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(Domains);
