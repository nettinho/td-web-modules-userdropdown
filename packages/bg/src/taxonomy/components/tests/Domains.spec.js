import React from "react";
import { shallow } from "enzyme";
import { Domains } from "../Domains";

describe("<Domains />", () => {
  const createDomain = jest.fn();
  const domainsLoading = false;
  const props = { createDomain, domainsLoading };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<Domains {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
