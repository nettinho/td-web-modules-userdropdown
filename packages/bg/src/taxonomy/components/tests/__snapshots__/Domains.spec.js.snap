// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Domains /> matches the latest snapshot 1`] = `
<React.Fragment>
  <Segment>
    <Header
      as="h2"
    >
      <Icon
        as="i"
        circular={true}
        name="sitemap"
      />
      <HeaderContent>
        <FormattedMessage
          id="navigation.glossary.taxonomy"
          values={Object {}}
        />
        <HeaderSubheader>
          <FormattedMessage
            id="domains.subheader"
            values={Object {}}
          />
        </HeaderSubheader>
      </HeaderContent>
    </Header>
    <Segment
      attached="bottom"
      loading={false}
    >
      <InjectIntl(Connect(DomainsActions)) />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/domains"
        />
        <Route
          exact={true}
          path="/domains/new"
        >
          <withRouter(InjectIntl(Connect(DomainForm)))
            handleSubmit={[MockFunction]}
          />
        </Route>
      </Switch>
    </Segment>
  </Segment>
</React.Fragment>
`;
